StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c

```
@@ -914,7 +914,7 @@ MmIplDispatchMmDrivers (
   //
   Size   = sizeof (CommunicateHeader);
   Status = Communicate (NULL, &CommunicateHeader, &Size);
-  ASSERT_EFI_ERROR (Status);
+  // ASSERT_EFI_ERROR (Status);

```

Not sure what to do with this.

---

This is PRed as is. Will see if others have different opinions.

---

For clarity, can you provide a simple comparison of design for this point?

---

The issue behind this is that the supervised Standalone MM hosts this handler in supervisor channel. Thus to invoke this handler, we need to indicate the given MM communicate call is targeting supervisor handler pool (we have that wrapped in separate PPI and protocol).

The call here will issue an MMI with user mode data populated and will end up with EFI_NOT_FOUND error because there is not handler for driver dispatching in the user space.

The removal of the asserts here is basically telling this IPL to ignore the error, and it will be handled later through a supv specific module.

---

I haven't followed the full code path, but would one of these options work?

- Make separate INF with the function that starts the call stack different for a "communicate" and "non-communicate" version of StandaloneMmIplPei.
- Setup a unique/appropriate status code return type that can bubble back up in this case and only assert if an error code and not that return type.

---

oh, i see :) then we will bury that in the ring3 broker layer. That is fine with me. Let me try that in a bit. Thanks~!

---

I ended up doing the second option, given the ring 3 broker is not dispatched yet. And the change will likely look like this: https://github.com/kuqin12/mu_feature_mm_supv/commit/348e63e0ebb10e7966184b5dc7b5d4e0b4d93cda

---

I think that's a little more preferable anyway to reduce platform integration overhead by avoiding two IPL INFs to pick from.

---

Okay, this change is updated.